### PR TITLE
Add Telephant

### PR DESCRIPTION
**Pull request by JBBgameich**
TODO:
 * Add description in appstream metadata

### CODE REVIEW

File: com.fribbledom.telephant.json [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                },
+                {
+                    "type": "git",
+                    "url": "https://github.com/davecgh/go-spew",
+                    "commit": "d8f796af33cc11cb798c1aaeb27a4ebc5099927d",
+                    "dest": "src/github.com/davecgh/go-spew"
+                },
+                {
+                    "type": "file",
+                    "path": "com.fribbledom.telephant.appdata.xml"
```
Comment (LeandroStanger) [inline com.fribbledom.telephant.json:126]: New appdata files should use metainfo instead. https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html

**Comment by LeandroStanger** [comment]
bot, build com.fribbledom.telephant

**Comment by barthalion** [comment]
bot, build com.fribbledom.telephant

File: com.fribbledom.telephant.json [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                {
+                    "type": "git",
+                    "url": "https://github.com/muesli/go-app-paths",
+                    "commit": "c184ea3ba2f517461749fca7a793d9609caff57d",
+                    "dest": "src/github.com/muesli/go-app-paths"
+                },
+                {
+                    "type": "git",
+                    "url": "https://github.com/muesli/telephant",
+                    "branch": "master",
```
Comment (barthalion) [inline com.fribbledom.telephant.json:91]: This needs to use specific commit/tag.

File: com.fribbledom.telephant.json [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                },
+                {
+                    "type": "git",
+                    "url": "https://github.com/davecgh/go-spew",
+                    "commit": "d8f796af33cc11cb798c1aaeb27a4ebc5099927d",
+                    "dest": "src/github.com/davecgh/go-spew"
+                },
+                {
+                    "type": "file",
+                    "path": "com.fribbledom.telephant.metainfo.xml"
```
Comment (barthalion) [inline com.fribbledom.telephant.json:126]: Needs to be submitted upstream when finished.

File: com.fribbledom.telephant.json [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+                }
+            ],
+            "build-options": {
+                "env": {
+                    "QT_PKG_CONFIG": "true"
+                }
+            },
+            "build-commands": [
+                ". /usr/lib/sdk/golang/enable.sh; export GOPATH=$PWD; go install -v -tags=no_env github.com/therecipe/qt/cmd/...",
```
Comment (barthalion) [inline com.fribbledom.telephant.json:135]: Do you actually need to source enable.sh every time?

File: com.fribbledom.telephant.json (hunk 2 above)
Comment (JBBgameich) [reply com.fribbledom.telephant.json:91]: Yeah, that's why there is a todo entry for it :)

File: com.fribbledom.telephant.json (hunk 4 above)
Comment (JBBgameich) [reply com.fribbledom.telephant.json:135]: At least everytime the go command is used. Apparently the environment is not preserved between the commands.

**Comment by LeandroStanger** [comment]
bot, build com.fribbledom.telephant

**Comment by LeandroStanger** [comment]
bot, build com.fribbledom.telephant

**Comment by JBBgameich** [comment]
Can you please give me a hint why the check for a 128x128px icon fails?
The command used by build bot to test it is:
```
test -f builddir/*/share/app-info/icons/flatpak/128x128/com.fribbledom.telephant.png
```

Accordidng to the build log the icon is exported:
```
Exporting share/icons/hicolor/128x128/apps/com.fribbledom.telephant.png
```

**Comment by barthalion** [comment]
Because desktop file name does not match app id: https://github.com/muesli/telephant/blob/master/assets/telephant.desktop

`Icon=` there should also follow app id.

**Comment by bilelmoussaoui** [comment]
bot, build com.fribbledom.telephant

**Comment by bilelmoussaoui** [comment]
Have you reached out to the upstream maintainer if they were interested in (co-)maintaining their app on Flathub?

**Comment by JBBgameich** [comment]
I'm working with the upstream maintainer on this, so I'm pretty sure they would like access to the repository once this is ready.

The current status of this pull request is that I'm waiting for a stable release upstream and a for me or @muesli to come up with a good description :)

**Comment by muesli** [comment]
Maintainer of Telepahnt here. Yes, I'm happy to (co-)maintain it on Flathub!
@JBBgameich Any idea why the latest build is still failing?

File: com.fribbledom.telephant.json [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+                ". /usr/lib/sdk/golang/enable.sh; export GOPATH=$PWD; cd $(go env GOPATH)/src/github.com/muesli/telephant; $(go env GOPATH)/bin/qtdeploy build desktop .",
+                "install -Dm755 src/github.com/muesli/telephant/deploy/linux/telephant $FLATPAK_DEST/bin/telephant",
+                "sed -i 's/Icon=telephant/Icon=com.fribbledom.telephant/g' -i src/github.com/muesli/telephant/assets/telephant.desktop",
+                "install -Dm644 src/github.com/muesli/telephant/assets/telephant.desktop $FLATPAK_DEST/usr/share/applications/com.fribbledom.telephant.desktop",
```
Comment (bilelmoussaoui) [inline com.fribbledom.telephant.json:140]:
```suggestion
                "install -Dm644 src/github.com/muesli/telephant/assets/telephant.desktop $FLATPAK_DEST/share/applications/com.fribbledom.telephant.desktop",
```

Comment (JBBgameich) [reply com.fribbledom.telephant.json]: Thanks! How could I miss this ...

**Comment by bilelmoussaoui** [comment]
bot, build com.fribbledom.telephant

**Comment by nedrichards** [comment]
OK, so other than the appdata description this seems fine to me

**Comment by JBBgameich** [comment]
Is the rc release fine for inclusion into flathub? If it is, I can hurry a bit with the description instead of waiting for the release :)

**Comment by nedrichards** [comment]
bot, build com.fribbledom.telephant

**Comment by nedrichards** [comment]
@JBBgameich depends how long before the stable release is out ;-) if it's a little while I can merge this to the beta branch and then you can do a stable update when available.

**Comment by JBBgameich** [comment]
@JBBgameich I'd like to leave this decision for @muesli, since he is the upstream maintainer and I followed his call for packagers on mastodon :)

**Comment by muesli** [comment]
I'd be happy to see this land on the beta branch first, while I'm trying to sort out the remaining issues. We can then move this to stable with Telephant's release being out.

Thanks for the work and feedback by everyone involved - I really appreciate your support! <3

**Pull request closed without merging**